includes/modules/searchandreplace/class-pb-search.php

```diff
@@ -14,11 +14,23 @@ class Search {
 	var $source = null;
 	var $error = null;
 
+	var $regex = false;
+	var $regex_error = null;
+
 	function name() {
 		return '';
 	}
 
+	function regex_error( $errno, $errstr, $errfile, $errline ) {
+		$this->regex_error = __( 'Invalid regular expression', 'pressbooks' ) . ': ' . preg_replace( '/(.*?):/', '', $errstr );
+	}
+
 	function search_and_replace( $search, $replace, $limit, $offset, $orderby, $save = false ) {
+		// escape potential backreferences when not in regex mode
+		if ( ! $this->regex ) {
+			$replace = str_replace( '\\', '\\\\', $replace );
+			$replace = str_replace( '$', '\\$', $replace );
+		}
 		$this->replace = $replace;
```

Comment:
Lets the `/e` modifier through which allows users to execute arbitrary code on the server?

@see https://bitquark.co.uk/blog/2013/07/23/the_unexpected_dangers_of_preg_replace